cpp/daal/src/data_management/roc_auc_score.cpp

```diff
@@ -17,90 +17,57 @@
 
 #include "data_management/data/internal/roc_auc_score.h"
 #include "data_management/data/numeric_table.h"
-#include "data_management/data/data_dictionary.h"
-#include "services/env_detect.h"
+#include "src/algorithms/service_kernel_math.h"
+#include "src/algorithms/service_sort.h"
 #include "src/externals/service_dispatch.h"
+#include "src/externals/service_math.h"
 #include "src/services/service_data_utils.h"
-#include "src/services/service_utils.h"
-#include "src/services/service_defines.h"
-#include "src/threading/threading.h"
-#include "src/data_management/service_numeric_table.h"
-#include "data_management/features/defines.h"
-#include "src/algorithms/service_error_handling.h"
-#include "src/externals/service_rng.h"
-#include "src/externals/service_rng_mkl.h"
 
 namespace daal
 {
 namespace data_management
 {
 namespace internal
 {
-/*
-template <typename DataType>
-struct Sum 
-{
-    double value;
-
-    Sum() : value(0) {}
-    Sum(Sum& s, tbb::split) : value(0) {}
-
-    void operator()(const tbb::blocked_range<typename std::vector<DataType>::iterator>& r) {
-        value = std::accumulate(r.begin(), r.end(), value);
-    }
-
-    void join(Sum& rhs) { value += rhs.value; }
-};
-*/
 template <typename DataType, daal::CpuType cpu>
 void calculateRankDataImpl(DataType * predictedRank, NumericTablePtr & prediction_numpy, const int& size)
 {
-    data_management::BlockDescriptor<DataType> numpyBlock;
-    prediction_numpy->getBlockOfRows(0, 1, data_management::readOnly, numpyBlock);
-    DataType * numpyPtr = numpyBlock.getBlockPtr();
-
-    std::pair<DataType, size_t>* v_sort = new std::pair<DataType, size_t>[size];
+    ReadRows<DataType, cpu> numpyBlock(prediction_numpy.get(), 0, 1);
+    const DataType * const numpyPtr = numpyBlock.get();
+    DataType * const values = new DataType[size];
```

Comment:
Use ```TArray``` with RAII instead of allocation by ```new```

Especially - I forgot to ```delete``` values array.